crates/shared/src/sources/balancer_v2/pool_fetching.rs

```diff
@@ -298,7 +299,7 @@ async fn create_aggregate_pool_fetcher(
     contracts: &BalancerContracts,
 ) -> Result<Aggregate> {
     let registered_pools = pool_initializer.initialize_pools().await?;
-    let fetched_block_number = registered_pools.fetched_block_number;
+    let fetched_block = registered_pools.fetched_block;
```

Comment:
Specifically, instead of making the `BalancerSubgraphClient` have a `web3` instance and fetch the block number, we can add here a:

```rust
let fetched_block_hash = web3.eth().block(registered_pools.fetched_block_number)...;
```

And then, pass the fetched block hash to the `create_internal_pool_fetcher` method. THis way, we don't need to change  any of the `PoolInitializing` implementations or the `RegisteredPools` struct - and we have a single place of logic where we query the node for the `fetched_block_hash` for Balancer pools.

Reply:
Indeed, thanks for noticing this. Fixed.